src/expressions/functions/builtInFunctions.node.ts

```diff
@@ -112,6 +112,84 @@ const fnHasChildren: FunctionDefinitionType = function(
 	});
 };
 
+const fnPath: FunctionDefinitionType = function(
+	_dynamicContext,
+	executionParameters,
+	_staticContext,
+	nodeSequence: ISequence
+) {
+	return zipSingleton([nodeSequence], ([nodeValue]: (Value | null)[]) => {
+		const node: ConcreteNode = nodeValue ? nodeValue.value : null;
+
+		if (!node) {
+			return sequenceFactory.empty();
+		}
+
+		let result = '';
+
+		function getChildIndex(child: ConcreteChildNode): number {
+			let i = 0;
+			const childNodeType = child.nodeType;
+			const childNodeName = child.nodeName;
+			while (child !== null) {
+				if (childNodeType === child.nodeType && childNodeName === child.nodeName) {
+					i++;
+				}
+				child = executionParameters.domFacade.getPreviousSibling(child);
+			}
+			return i;
+		}
+
+		for (
+			let ancestor: ConcreteNode = node;
+			ancestor;
+			ancestor = executionParameters.domFacade.getParentNode(
+				ancestor as ConcreteChildNode
+			) as ConcreteNode
+		) {
+			if (
```

Comment:
Could you make this the 'exit condition' of the loop? like:

```
for (let ancestor: ConcreteNode = node;
domFacade.getParentNode(ancestor) !== null;
ancestor = ...) {
```

And move the returning part to after the loop?

This way the loop gets more readable~

Reply:
done